cmd/main.go

```diff
@@ -154,43 +153,25 @@ func main() {
 		}
 	}()
 
-	// update cache every 2 seconds
-	go func() {
-		for {
-			if _, err := configuration.ForceLoadRegistrationServiceConfig(cl); err != nil {
-				log.Error(nil, err, "failed to update the configuration cache")
-			}
-			time.Sleep(2 * time.Second)
-		}
-	}()
-
-	gracefulShutdown(configuration.GracefulTimeout, regsvcSrv.HTTPServer(), regsvcMetricsSrv, proxySrv, proxyMetricsSrv)
+	gracefulShutdown(ctx, configuration.GracefulTimeout, regsvcSrv.HTTPServer(), regsvcMetricsSrv, proxySrv, proxyMetricsSrv)
 }
 
-func gracefulShutdown(timeout time.Duration, hs ...*http.Server) {
-	// For a channel used for notification of just one signal value, a buffer of
-	// size 1 is sufficient.
-	stop := make(chan os.Signal, 1)
-
-	// We'll accept graceful shutdowns when quit via SIGINT (Ctrl+C) or SIGTERM
-	// (Ctrl+/). SIGKILL, SIGQUIT will not be caught.
-	signal.Notify(stop, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
-	sigReceived := <-stop
-	log.Infof(nil, "Signal received: %+v", sigReceived.String())
-
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+func gracefulShutdown(ctx context.Context, timeout time.Duration, hs ...*http.Server) {
+	<-ctx.Done()
```

Comment:
is this actually required ? I see the reading from the Done channel is already happening in the `Shutdown` function.

Reply:
yes, it is - this is the notifier (gate keeper) that tells us when the kill signal was send and when we should start the graceful shutdown.

Reply:
ok, I see I thought the one in the select statement would suffice but I was wrong. Thanks.

Reply:
which select statement :thinking: ?

Reply:
Ah, you mean in the `Shutdown` method - the select happens at the end of the shutdown logic when most of the stuff is already shut down and the logic closes the idled connections.